### PR TITLE
Whenever we adjust sales, adjust stock

### PR DESCRIPTION
**Pull request by Camwyn**
🎫 https://central.tri.be/issues/102634

### CODE REVIEW

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | Don't leave error_log lying around | [RSVP.php:1769](https://github.com/moderntribe/event-tickets/blob/62d29a6b75edd91cc3b6ece23d5de60bbbad187b/src/Tribe/RSVP.php#L1769)
:x: | There must be a single space after the first parenthesis | [RSVP.php:1769](https://github.com/moderntribe/event-tickets/blob/62d29a6b75edd91cc3b6ece23d5de60bbbad187b/src/Tribe/RSVP.php#L1769)
:x: | There must be a single space before the last parenthesis | [RSVP.php:1769](https://github.com/moderntribe/event-tickets/blob/62d29a6b75edd91cc3b6ece23d5de60bbbad187b/src/Tribe/RSVP.php#L1769)
:x: | There must be a single space before the last parenthesis | [RSVP.php:1781](https://github.com/moderntribe/event-tickets/blob/62d29a6b75edd91cc3b6ece23d5de60bbbad187b/src/Tribe/RSVP.php#L1781)
:warning: | The method parameter $order_id is never used | [RSVP.php:1822](https://github.com/moderntribe/event-tickets/blob/62d29a6b75edd91cc3b6ece23d5de60bbbad187b/src/Tribe/RSVP.php#L1822)
:warning: | The method parameter $attendee_order_status is never used | [RSVP.php:1822](https://github.com/moderntribe/event-tickets/blob/62d29a6b75edd91cc3b6ece23d5de60bbbad187b/src/Tribe/RSVP.php#L1822)
:warning: | The method parameter $ticket_id is never used | [RSVP.php:1822](https://github.com/moderntribe/event-tickets/blob/62d29a6b75edd91cc3b6ece23d5de60bbbad187b/src/Tribe/RSVP.php#L1822)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | There must be a single space before the last parenthesis | [RSVP.php:1780](https://github.com/moderntribe/event-tickets/blob/373ea2364e77c8f6284aadb7ae67f1441f99e64f/src/Tribe/RSVP.php#L1780)
:warning: | The method parameter $order_id is never used | [RSVP.php:1821](https://github.com/moderntribe/event-tickets/blob/373ea2364e77c8f6284aadb7ae67f1441f99e64f/src/Tribe/RSVP.php#L1821)
:warning: | The method parameter $attendee_order_status is never used | [RSVP.php:1821](https://github.com/moderntribe/event-tickets/blob/373ea2364e77c8f6284aadb7ae67f1441f99e64f/src/Tribe/RSVP.php#L1821)
:warning: | The method parameter $ticket_id is never used | [RSVP.php:1821](https://github.com/moderntribe/event-tickets/blob/373ea2364e77c8f6284aadb7ae67f1441f99e64f/src/Tribe/RSVP.php#L1821)

> via jenkins/codesniffer-phpcs

**Comment by Camwyn** [comment]
@bordoni I'd really appreciate your 👁 on this, as you know the stock/capacity system better than most! 😄

File: src/Tribe/RSVP.php [hunk 1]
```diff
@@ -1716,37 +1716,111 @@ public function updated_messages( $messages ) {
 	}
 
 	/**
-	 * Updates the product sales if old and new order stati differ in stock size.
+	 * Determine if the order stati are different (and we need to update the meta)
+	 * @param $order_id
```
Comment (bordoni) [inline src/Tribe/RSVP.php]: Add `@since`

File: src/Tribe/RSVP.php [hunk 2]
```diff
@@ -1716,37 +1716,111 @@ public function updated_messages( $messages ) {
...
+	 * @param int    $order_id
+	 * @param string $attendee_order_status
+	 * @param int    $ticket_id
+	 */
+	public function update_sales_and_stock_by_order_status( $order_id, $attendee_order_status, $ticket_id ) {
+		$sales_diff = $this->find_updated_sales_or_stock_value( $order_id, $attendee_order_status, $ticket_id, 'total_sales' );
+
+		// it's all or none here...
+		if ( false === $sales_diff ) {
 			return;
```
Comment (bordoni) [inline src/Tribe/RSVP.php]: Let's `return false` here

File: src/Tribe/RSVP.php [hunk 3]
```diff
@@ -1716,37 +1716,111 @@ public function updated_messages( $messages ) {
...
 		}
 
-		$previous_order_status_stock_size = $rsvp_options[ $previous_order_status ]['decrease_stock_by'];
-		$attendee_order_status_stock_size = $rsvp_options[ $attendee_order_status ]['decrease_stock_by'];
+		$stock_diff = $this->find_updated_sales_or_stock_value( $order_id, $attendee_order_status, $ticket_id, '_stock' );
 
-		if ( $previous_order_status_stock_size == $attendee_order_status_stock_size ) {
+		// it's all or none here...
+		if ( false === $stock_diff ) {
 			return;
```
Comment (bordoni) [inline src/Tribe/RSVP.php]: Let's `return false` here

File: src/Tribe/RSVP.php [hunk 4]
```diff
@@ -1716,37 +1716,111 @@ public function updated_messages( $messages ) {
...
 		}
 
-		$sales = (int) get_post_meta( $ticket_id, 'total_sales', true );
-		$diff  = $attendee_order_status_stock_size - $previous_order_status_stock_size;
+		// these should NEVER be updated separately - if one goes up the other must go down and vice versa
+		update_post_meta( $ticket_id, 'total_sales', $sales_diff );
+		update_post_meta( $ticket_id, '_stock', $stock_diff );
```
Comment (bordoni) [inline src/Tribe/RSVP.php]: Lets return the combined status of these updates. Just to make easier to test this later

File: src/Tribe/RSVP.php [hunk 5]
```diff
@@ -1716,37 +1716,111 @@ public function updated_messages( $messages ) {
...
+
+		return array(
+			'previous_stock_size' => $rsvp_options[ $previous_order_status ]['decrease_stock_by'],
+			'attendee_stock_size' => $rsvp_options[ $attendee_order_status ]['decrease_stock_by'],
+		);
+	}
+
+	/**
+	 * Get updated value for stock or sales, based on order status
+	 *
```
Comment (bordoni) [inline src/Tribe/RSVP.php]: Let's add `@since`

File: src/Tribe/RSVP.php (hunk 4 above)
Comment (Camwyn) [reply src/Tribe/RSVP.php]: Hm 🤔 as an array? Something like:
```
return array(
    update_post_meta( $ticket_id, 'total_sales', $sales_diff ),
    update_post_meta( $ticket_id, '_stock', $stock_diff ),
);
```
❓

Comment (Camwyn) [reply src/Tribe/RSVP.php]: First instinct is boolean since we're returning false (now) earlier?

**Comment by Camwyn** [comment]
Also, added some stock tests at the last minute to mirror the sales tests!

**Pull request merged**